### ISSUE TITLE
Replace syslog exporter with syslog exporter from opentelemetry-collector-contrib

### ISSUE THREAD

**Issue by kkujawa-sumo** [SumoLogic/sumologic-otel-collector#1200]
[Syslog exporter](https://github.com/SumoLogic/sumologic-otel-collector/tree/main/pkg/exporter/syslogexporter) which is in this repository should be replaced with [syslog exporter from opentelemetry-collector-contrib](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/exporter/syslogexporter).

Configuration options in [syslog exporter from opentelemetry-collector-contrib](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/exporter/syslogexporter) are different and slightly different code.

[syslog exporter from opentelemetry-collector-contrib](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/exporter/syslogexporter) has `development` stability, to move this exporter to alpha following tasks needs to be done:
- https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/21245
- https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/21244
- https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/21242

**Comment by kkujawa-sumo** [comment]
Example configuration for [sumologic version of syslog exporter](https://github.com/SumoLogic/sumologic-otel-collector/tree/main/pkg/exporter/syslogexporter):
```yaml
  syslog:
    protocol: tcp
    port: 514
    endpoint: 127.0.0.1
    format: rfc5424
    tls:
      ca_file: ca.pem
      cert_file: cert.pem
      key_file: key.pem
```

Example configuration for [upstream version of syslog exporter](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/exporter/syslogexporter):
```yaml
  syslog:
    network: tcp
    port: 514
    endpoint: 127.0.0.1
    protocol: rfc5424
    tls:
      ca_file: ca.pem
      cert_file: cert.pem
      key_file:  key.pem
```
so to switch to upstream version of syslog exporter it is necessary to replace:
- `protocol` key with `network`key
-  `format` key with `protocol` key

**Comment by kkujawa-sumo** [comment]
Syslog exporter will be available in v0.89 release of opentelemetry-collector-contrib, related pull request: https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/28902
